### ISSUE TITLE
Add Timestamp to stream OSD @ Ingest

### ISSUE THREAD

**Issue by marcoherbst** [evercam/ex_nvr#187]
Similar to undistorting. 

Using different cameras we're getting different qualities of timestamp.

Let's offer the option to ingest an image with no timestamp and to add the NVR timestamp.

Benefit 1: It will be more beautiful and consistent accross cameras
Benefit 2: We remove one possible source of confusion (a camera with the wrong time settings).

**Comment by marcoherbst** [comment]
https://github.com/membraneframework/membrane_ffmpeg_video_filter_plugin

TextOverlay element is implemented, based on [ffmpeg drawtext filter](https://ffmpeg.org/ffmpeg-filters.html#drawtext-1). This element enables adding text on top of given raw video frames.

**Comment by oussamabonnor1** [comment]
@gBillal @halimb isnt this done here ? [Add x-timestamp header to the live snapshot response](https://github.com/evercam/ex_nvr/pull/654)

**Comment by halimb** [comment]
@oussamabonnor1 no, the "x-timestamp" header is just an HTTP header.
I think this issue is about:
1- configure camera to not overlay timestamp text on the video stream.
2- make ex_nvr render an overlay timestamp on the video stream (using ffmpeg)

**Comment by halimb** [comment]
**Personal opinion on this issue**
Although we haven't tried it, I agree with Billal that it would be costly to burn the timestamp on the RTSP stream (needs decoding / re-encoding). In addition to the hardware cost, it will inevitably introduce an additional delay for the consumers of the feed.

**Favorite option:**
Get as close to the camera as possible (let the camera be the source of truth)

Either using:
- **EXIF data** (when available + depends on vendors)
- (costly, but efficient) running **OCR every few seconds** to calculate the drift from the camera timestamp and adjust ExNVR's internal clock.

**Least favorite:**
If we really want to have ex_nvr decide and encode timestamps, we may go for a soft-overlay (i.e.: **subtitles in an SRT file**, muxed with the MP4) 
we can use that file later for burning the timestamps when a snapshot is requested, or when an HLS stream is started.

**Comment by marcoherbst** [comment]
Yes to all of this, including the statement that we should not be decoding/encoding the camera stream - for now. 

But, in the not too distant roadmap, 2 things are likely to happen:
a) We may be decoding/encoding in order to do computer vision on the edge.
b) We may be supporting USB or non-RTSP video inputs that provide raw frames, so, no decoding required, only encoding. In that story, the equation looks different.

It's ok to close this or to put  it on the long-finger. It's valid, but not now, as I see it.
